km52/Larionova_Yuliia/laboratory1.sql

```diff
@@ -1,2 +1,135 @@
 -- LABORATORY WORK 1
 -- BY Larionova_Yuliia
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+вставляти дані до таблиць
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+create user larionova identified by pass
+default tablespace "USERS"
+temporary tablespace "TEMP";    
+
+ALTER USER larionova QUOTA 100M ON USERS;
+
+GRANT "CONNECT" TO larionova ;
+
+GRANT INSERT ANY TABLE TO larionova ;
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Класна кімната має парти та стільці.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+create table chairs(
+    chair_name varchar2(40) not null
+);
+
+alter table chairs
+    ADD CONSTRAINT chairs_pk PRIMARY KEY (chair_name);
+
+
+create table desks(
+    desk_name varchar2(40) not null
+);
+
+alter table desks
+    ADD CONSTRAINT desks_pk PRIMARY KEY (desk_name);
+
+create table classroom(
+    chairs_fk varchar2(40),
+    desks_fk varchar2(40),
+    desks_amount number(3),
+    chairs_amount number(3)
+);
+
+ALTER TABLE  classroom
+  ADD CONSTRAINT classroom_pk PRIMARY KEY (chairs_fk,desks_fk);  
+  
+ALTER TABLE  classroom
+  ADD CONSTRAINT chairs_fk FOREIGN KEY (chairs_fk) REFERENCES chairs (chair_name);
+  
+ALTER TABLE  classroom
+  ADD CONSTRAINT deks_fk FOREIGN KEY (desks_fk) REFERENCES desks (desk_name);
+
+
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+GRANT CREATE ANY TABLE TO larionova ;
+GRANT ALTER ANY TABLE TO larionova ;
+GRANT SELECT ANY TABLE TO larionova ;
+GRANT UPDATE ANY TABLE TO larionova;
```

Comment:
~~`GRANT UPDATE ANY TABLE TO larionova;`~~

Reply:
`GRANT INSERT ANY TABLE TO larionova;`

Reply:
> `GRANT INSERT ANY TABLE TO larionova;`

It's not necessary since  I already granted insert in the first task.
https://github.com/LarionovaJulia/dbis/blob/39c64cc8228f89485692e1d7fdba3689d6a6f578/km52/Larionova_Yuliia/laboratory1.sql#L18